### PR TITLE
Simple version

### PR DESCRIPTION
**Pull request by xjamundx**
Mostly a :trollface:  request to add support for some basic checking of common fields, but this actually works pretty well so :)

### CODE REVIEW

**Comment by xjamundx** [comment]
Some of the configs validate and some don't. Needs work :)

File: src/index.js [hunk 1]
```diff
@@ -1,13 +1,55 @@
...
+        if (allowed[key]) {
+          isGood = true;
+          if (typeof allowed[key] === 'function') {
+            isGood = allowed[key](config[key]);
+          } else if (typeof allowed[key] === 'object') {
+            // console.log(allowedThings[key], config[key])
+            var bads = getBadConfig(config[key], allowed[key]);
+            isGood = (bads.length === 0);
+            if (!isGood) {
+              console.log(allowed[key], config[key])
```
Comment (xjamundx) [inline src/index.js]: oh we don't need this

File: src/index.js [hunk 2]
```diff
@@ -1,13 +1,55 @@
...
+        }
+        return !isGood;
+      }) || [];
 }
 
+module.exports = function(config, options) {
+  var badConfig = getBadConfig(config, allowedThings).forEach(function(key) {
+    console.error("Unexpected key \"" + key + "\" found in config.");
+  });
+  if (options.strict && badConfig.length) {
```
Comment (xjamundx) [inline src/index.js]: I think this is a good idea if your things doesn't have it already "strict mode"

File: tests/configs/bad/typos.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+	sentry: "go",
```
Comment (xjamundx) [inline tests/configs/bad/typos.js]: Eventually we'll add some kind of fancy "you may have meant" feature to get this right (should be "entry")

**Comment by xjamundx** [comment]
Core request here is validation on the following paths + unvalidated path warnings:
```
  "entry": isStringOrNotEmptyObject,
  "module": { loaders: Array.isArray },
  "output": { path: isString, filename: isString },
  "plugins": Array.isArray,
  "devtool": isString,
  "context": isString,
  "resolve": {
    modulesDirectories: Array.isArray,
    extensions: Array.isArray,
    alias: isntEmpty
  }
```

**Comment by kentcdodds** [comment]
Thanks for this! I think we're set now!

**Pull request closed without merging**